indexer/Indexer.cc

```diff
@@ -413,6 +405,49 @@ void TuIndexer::saveNestedNameSpecifier(
   }
 }
 
+void TuIndexer::saveRecordDecl(const clang::RecordDecl &recordDecl) {
+  this->saveTagDecl(recordDecl);
+  // Superclass declarations will be visited during Visit*TypeLoc methods.
+}
+
+void TuIndexer::saveRecordTypeLoc(const clang::RecordTypeLoc &recordTypeLoc) {
+  this->saveTagTypeLoc(recordTypeLoc);
+}
+
+void TuIndexer::saveTagDecl(const clang::TagDecl &tagDecl) {
+  auto optSymbol = this->symbolFormatter.getTagSymbol(tagDecl);
+  if (!optSymbol.has_value()) {
+    return;
+  }
+  auto symbol = optSymbol.value();
+
+  if (!tagDecl.isThisDeclarationADefinition()) {
+    // Forward declarations should be marked as references. In the future,
+    // we should add an extra role to SCIP for forward declarations.
+    this->saveReference(symbol, tagDecl.getLocation());
+    return;
+  }
+
+  llvm::SmallVector<std::string, 4> docComments{};
```

Comment:
Why 4 and not, say, 1? Are these lines of comments? I feel like 99.999% of things will have zero doc comments and 99.99999% of things will have zero or one. (I assume SmallVector can spill to the heap if there's more than that...)

Popping up a level, presumably you're iterating over something in tryGetDocComment, why not simply adapt and expose that iterator? This calling code would be more succinct without the vector etc. All it wants to do is iterate over them.

Reply:
Changed the underlying method to return a `vector` directly.

> Popping up a level, presumably you're iterating over something in tryGetDocComment, why not simply adapt and expose that iterator?

The Clang API being called returns a owned `std::vector`, so I can't return an iterator from `tryGetDocComment`.